### ISSUE TITLE
Incorrect 'content-type' when attachment is embedded

### ISSUE THREAD

**Issue by alex4584** [Jericho/StrongGrid#242]
Hi,

I'm having issues trying to parse a meeting request type of e-mail sent from outlook. I end up with an attachment that has the content type "text/plain" instead of "text/calendar" and name, filename, contentId are null.
When sending the same type of e-mail from gmail, i encountered the same behaviour except that one extra attachment is parsed which have all the data correctly set.

Thanks

**Comment by Jericho** [comment]
I’m afraid that I will need more information in order to diagnose your situation. Can you provide more information such as code sample and an example of such an email.

**Comment by Jericho** [comment]
One thing that would help is if you could get me a copy of the data posted to you by SendGrid. Here's a code sample that will allow you to save this data to a file:

```csharp
using (var body = new MemoryStream())
{
  Request.Body.CopyTo(body);
  body.Position = 0;

  using (var file = new FileStream("C:\\Temp\\Inbound.bin", FileMode.Create, FileAccess.Write))
  {
    byte[] bytes = new byte[body.Length];
    body.Read(bytes, 0, (int)body.Length);
    file.Write(bytes, 0, bytes.Length);
    body.Close();
  }
}
```

**Comment by alex4584** [comment]
Sure, please see below.

![meeting](https://user-images.githubusercontent.com/42472668/44325944-dad5e000-a462-11e8-931f-ff984f0e198a.PNG)

The content of the e-mail could be empty or not, it doesn't matter.

![empty](https://user-images.githubusercontent.com/42472668/44325950-df9a9400-a462-11e8-80a1-b3ae70c78a18.png)

This is what you will find related to the attachment:

Content-Disposition: form-data; name="attachment1"; filename="msg-26831-197.txt"
Content-Type: text/plain

Content-Disposition: form-data; name="attachment-info"
{"attachment1":{"charset":"utf-8","type":"text/calendar"}}

**Comment by Jericho** [comment]
I can reproduce the scenario you are describing by following the steps you indicated

Here is the relevant section in the data posted to me by SendGrid:

![image](https://user-images.githubusercontent.com/112710/44354563-32cb1180-a478-11e8-92a2-39623253e141.png)

As you can see, Outlook has embedded the content of the calendar item as plain text which explains why the `content-type` is `text/plain`. I am speculating that gmail attaches a `.ics` calendar file rather than embedding the content of the calendar item which would explain why you get a different result when sending from gmail.

However, as you pointed out, the `attachment-info` section seems to contain a value that more accurately describes the content:
![image](https://user-images.githubusercontent.com/112710/44355023-63f81180-a479-11e8-8bfa-145055604f59.png)

Currently, StrongGrid favors the content-type of the actual attachment (which is `text/plain` in this example) over the content type in the `attachment-info` section (which is `text/calendar` in this example) which explains why the content-type is not the value that you expect when the content is embedded in the email.

So, in conclusion, I need to change the logic to favor the content-type in the attachment-info if appropriate.
